DOC: mention x-axis scaling for waveform()

hagenw opened the pull request:
I tested the new `audplot.waveform()` code and it works as expected, but I also had to change code where I was setting the `xlim` after plotting as the x-axis now no longer reflects the actual signal length.
This pull request updates the docstring of `audplot.waveform()` mentioning that the x-axis might be scaled.

![image](https://user-images.githubusercontent.com/173624/205586665-b957a479-2eb2-4f53-8413-e443164bca6b.png)

frankenjoe commented:
Do you really think we need to add this to the documentation? At least it's a bit confusing now, as it sounds like you can get the length of the signal by calling `ax.get_xlim()`, but with "actual length" you mean the length of the downsampled signal, correct? Generally, why do you think a user will try to get the signal size from the axis?

hagenw commented:
If a user wants to specify a custom `xlim`, like show me the signal from 20% to 70% she could achieve this by setting `xlim` based on the number of samples in the signal. This no longer works now, and you first need to get the maximum value of the x-axis with `ax.get_xlim()[1]` in order to calculate what 20% and 70% would be.

But I'm also ok with removing the last sentence I added and not mention `ax.get_xlim()`.

frankenjoe commented:
Ok if it's really a problem then we should look for a proper solution and properly set the x axis, e.g. to the first index of the according bin.

frankenjoe commented:
Something like `(np.arange(len(x_split))) * (samples / len(x_split))` should work

hagenw commented:
> (np.arange(len(x_split))) * (samples / len(x_split))

Very good idea.

I changed the code accordingly and added a test for it. Now the behavior for `xlim` is also the same for short (not downsampled) and long signals

frankenjoe commented on tests/test_api.py:
```diff
@@ -52,4 +52,8 @@ def test_waveform():
+        ax = plt.gca()
+        xlim = ax.get_xlim()
+        assert xlim[0] == 0
+        assert xlim[1] == n
```
This currently only works because you have the case `n < pixels`. With a larger `n` it fails, but I think that is only because of the following line you added: `samples = x_axis[-1] + 1`. When I removed it, then the test was also passing for larger values of `n`.

hagenw replied:
This is strange, I checked locally, and for me it has `pixels=800`, which means the 1200 samples test case is already > pixels.

frankenjoe commented on audplot/core/api.py:
```diff
@@ -818,11 +818,12 @@ def waveform(
         x_max = [x.max() for x in x_split]
         x_min = [x.min() for x in x_split]
+        x_axis = (np.arange(len(x_split))) * (samples / len(x_split))
+        samples = x_axis[-1] + 1
```
Why do we need this line? Since we later set `xlim = (0, samples)` it means we possibly end up with a different size again.

hagenw replied:
You are right, I removed that line.
